lib/azure/armrest/insights/metrics_service.rb

```diff
@@ -0,0 +1,59 @@
+module Azure
+  module Armrest
+    module Insights
+      class MetricsService < ArmrestService
+        # Creates and returns a new MetricsService object.
+        #
+        def initialize(armrest_configuration, options = {})
+          options['api_version'] = '2014-04-01' # Must hard code for now
+          super(armrest_configuration, 'metricDefinitions', 'Microsoft.Insights', options)
+        end
+
+        # Return the metric definitions for the given +provider+, +resource_type+,
+        # and +resource_name+ for +resource_group+. You may pass a :filter option as well.
+        #
+        # Example:
+        #
+        #   metrics = Azure::Armrest::Insights::MetricsService.new(conf)
+        #
+        #   metrics.list('Microsoft.SQL', 'servers', 'myServer/databases/myDB', 'mygroup')
+        #   metrics.list('Microsoft.Compute', 'virtualMachines', 'myVm', 'mygroup')
+        #
+        def list(provider, resource_type, resource_name, resource_group = nil, options = {})
+          resource_group ||= configuration.resource_group
+
+          raise ArgumentError, "no resource group provided" unless resource_group
+
+          url = build_url(provider, resource_type, resource_name, resource_group, options)
+
+          response = rest_get(URI.escape(url))
```

Comment:
Because of the $filter syntax. Here's a URL without escape:

```
"https://management.azure.com/subscriptions/xxx/resourceGroups/yyy/providers/Microsoft.Compute/virtualMachines/zzz/metricDefinitions?api-version=2014-04-01&$filter=name.value eq '\\Processor(_Total)\\% Processor Time'"
```
Same URL now escaped:
```
"https://management.azure.com/subscriptions/xxx/resourceGroups/yyy/providers/Microsoft.Compute/virtualMachines/zzz/metricDefinitions?api-version=2014-04-01&$filter=name.value%20eq%20'%5CProcessor(_Total)%5C%25%20Processor%20Time'"
```
But, this may not be explicitly necessary. It looks to me like rest-client is escaping the parameters automatically, but I can double check. If not, I think we should escape the url's in our core methods.